tests/test_arrange.py

```diff
@@ -49,77 +51,85 @@ def test_CleanInfilepath_Underscores_Match():
     assert arrange.clean_filename("dicom__.zip") == "dicom"
 
 
-def test_PrepareDcm2niixInput_ZipArchive_MatchValidDataset():
-
-    versions = ["dicom_nested", "dicom_single", "parrec_single"]
-
-    for version in versions:
+@pytest.mark.parametrize("version, ext", [
+    ("dicom_nested_one_level", "zip"),
+    ("dicom_nested_two_levels", "zip"),
+    ("dicom_nested_uneven", "zip"),
 
-        tmp_dir = f"{ASSETS_DIR}/tmp"
-        shutil.os.mkdir(tmp_dir)
-        infile = f"{ASSETS_DIR}/{version}.zip"
+    ("dicom_single", "tgz"),
+    ("parrec_single", "tgz")
+])
+def test_PrepareDcm2niixInput_Zip_Tar_Archive_MatchValidDataset(tmpdir, version, ext):
 
-        arrange.prepare_dcm2niix_input(infile, False, tmp_dir)
-        valid_dir = f"{ASSETS_DIR}/valid_dataset/{version}"
-        out = filecmp.dircmp(tmp_dir, valid_dir)
-
-        assert out.left_only == []
-        assert out.right_only == []
+    # define input file, define and populate new_dir
+    infile = f"{ASSETS_DIR}/{version}.{ext}" #
 
-        shutil.rmtree(tmp_dir)
+    # define old dir
+    old_dir = f"{ASSETS_DIR}/valid_dataset/{version}" # corresponding unzipped dir
 
+    # define and populate new dir
+    if os.path.exists(f"{ASSETS_DIR}/tmp"):
+        shutil.rmtree(f"{ASSETS_DIR}/tmp")
+    os.mkdir(f'{ASSETS_DIR}/tmp')
+    new_dir = arrange.prepare_dcm2niix_input(infile, False, f"{ASSETS_DIR}/tmp")
```

Comment:
Can we move away from creating/remove `tmp` dir here and use the pytest `tmpdir` instead?

Reply:
👍